src/github-cli/install.sh

```diff
@@ -124,138 +124,85 @@ check_packages() {
-            fi
+    local prefix="${1:-}"
+    git ls-remote --tags --sort=-version:refname https://github.com/cli/cli.git | while read -r _ ref; do
+        version="${ref#refs/tags/v}"
+        if [[ $version != *-* && ( -z $prefix || "${version}." == "${prefix}."* ) ]]; then
+            printf "%s\n" "$version"
         fi
-    set -e
+    done
 }
 
-# Fall back on direct download if no apt package exists
-# Fetches .deb file to be installed with dpkg
+# For each version, fetch a .deb file from releases and install the first available one with dpkg
 install_deb_using_github() {
-    check_packages wget
-    arch=$(dpkg --print-architecture)
-
-    find_version_from_git_tags CLI_VERSION https://github.com/cli/cli
-    cli_filename="gh_${CLI_VERSION}_linux_${arch}.deb"
-
-    mkdir -p /tmp/ghcli
-    pushd /tmp/ghcli
-    wget https://github.com/cli/cli/releases/download/v${CLI_VERSION}/${cli_filename}
-    exit_code=$?
-    set -e
-    if [ "$exit_code" != "0" ]; then
-        # Handle situation where git tags are ahead of what was is available to actually download
-        echo "(!) github-cli version ${CLI_VERSION} failed to download. Attempting to fall back one version to retry..."
-        find_prev_version_from_git_tags CLI_VERSION https://github.com/cli/cli
-        wget https://github.com/cli/cli/releases/download/v${CLI_VERSION}/${cli_filename}
-    fi
-
-    dpkg -i /tmp/ghcli/${cli_filename}
-    popd
+    local arch="$(dpkg --print-architecture)"
+    local version download_url
+    for version; do
+        download_url="https://github.com/cli/cli/releases/download/v${version}/gh_${version}_linux_${arch}.deb"
+        wget -P /tmp/ghcli "$download_url" || continue
+        dpkg -i "/tmp/ghcli/$(basename "$download_url")"
+        break
+    done
     rm -rf /tmp/ghcli
 }
 
-export DEBIAN_FRONTEND=noninteractive
-
-# Install curl, apt-transport-https, curl, gpg, or dirmngr, git if missing
-check_packages curl ca-certificates apt-transport-https dirmngr gnupg2
-if ! type git > /dev/null 2>&1; then
-    check_packages git
+if [ -n "$TEST_EXPORT" ]; then
+    # print function named by TEST_EXPORT and exit
+    declare -f "$TEST_EXPORT"
+    exit 0
 fi
 
-# Soft version matching
-if [ "${CLI_VERSION}" != "latest" ] && [ "${CLI_VERSION}" != "lts" ] && [ "${CLI_VERSION}" != "stable" ]; then
-    find_version_from_git_tags CLI_VERSION "https://github.com/cli/cli"
-    version_suffix="=${CLI_VERSION}"
-else
-    version_suffix=""
+check_packages ca-certificates
+type -P git >/dev/null || check_packages git
+
+is_latest="false"
+
+case "$CLI_VERSION" in
+none )
+    # opt out of installing gh
+    exit 0
+    ;;
+latest | lts | stable | current )
+    # find two latest versions
+    versions=( $(find_version_from_git_tags | head -n2) )
+    is_latest="true"
+    ;;
+*.*.* )
+    # the exact version is specified explicitly
+    versions=( "$CLI_VERSION" )
+    ;;
+* )
```

Comment:
Can we also add another case for `*.*` ? that's something users might use as well.

Reply:
Good point, but the case for `*.*` is already handled by the general fallback case `*`.